README.md

```diff
@@ -74,127 +73,135 @@ To install Prince of Versions from the Swift Package Manager, you should:
-
-Key-value pairs under `"meta"` key are optional metadata of which any amount can be sent accompanying the required fields.
+For JSON file details and formatting, read [JSON specification](JSON.md).
 
 ## Usage
 
 ### Loading from network resource
 
-1. Getting all data
-
-    ```swift
-    let url = URL(string: "https://pastebin.com/raw/ZAfWNZCi")
-    PrinceOfVersions().loadConfiguration(from: url) { response in
-        switch response.result {
-        case .success(let info):
-            print("Minimum version: ", info.minimumRequiredVersion)
-            print("Installed version: ", info.installedVersion)
-            print("Is minimum version satisfied: ", info.isMinimumVersionSatisfied)
-            print("Notification type: ", info.notificationType)
-
-            if let latestVersion = info.latestVersion {
-                print("Is minimum version satisfied: ", latestVersion)
-            }
-        case .failure(let error):
-            print(error.localizedDescription)
-        }
-    }
-    ```
-
-2. Automatic handling update frequency
-
-    ```swift
-    let url = URL(string: "https://pastebin.com/raw/ZAfWNZCi")
-    PrinceOfVersions().checkForUpdates(
-        from: url,
-        newVersion: { (latestVersion, isMinimumVersionSatisfied, metadata) in
-            ...
-        },
-        noNewVersion: { (isMinimumVersionSatisfied, metadata) in
-            ...
-        },
-        error: { error in
-            ...
-        })
-    ```
+#### Getting all data
+
+  ```swift
+  let princeOfVersionsURL = URL(string: "https://pastebin.com/raw/0MfYmWGu")!
+
+  PrinceOfVersions.checkForUpdates(from: princeOfVersionsURL, completion: { [unowned self] response in
+      switch response.result {
+      case .success(let updateResultData):
+          print("Update version: \(updateResultData.updateVersion)")
+          print("Installed version: \(updateResultData.updateInfo.installedVersion)")
+          print("Update status: \(updateResultData.updateStatus)")
+      case .failure:
+          // Handle error
+          break
+      }
+  })
+  ```
+
+#### Adding-requirements
+
+For each requirement key listed in a configuration, there should exist a requirement check closure. If you don't provide it, the requirement will be considered as not met, and the whole configuration will be discarded.
+
+Here is the example of how to add requirement check closures.
+
+  ```swift
+  let options = PoVRequestOptions()
+
+  options.addRequirement(key: "region") { (value) -> Bool in
+      guard let value = value as? String else { return false }
+      // Check OS localisation
+      return value == "hr"
+  }
+
+  options.addRequirement(key: "bluetooth") { (value) -> Bool in
+      guard let value = value as? String else { return false }
+      // Check device bluetooth version
+      return value.starts(with: "5")
```

Comment:
If you have changed this in example app (#31 ), please update this readme as well :)